core/api/src/services/openai/index.ts

```diff
@@ -0,0 +1,114 @@
+import OpenAI, { OpenAIError } from "openai"
+
+import { env } from "@/config/env"
+import { sleep } from "@/utils"
+import { UnknownDomainError } from "@/domain/shared"
+
+let openai: OpenAI
+
+if (env.OPENAI_API_KEY) {
+  openai = new OpenAI({
+    apiKey: env.OPENAI_API_KEY,
+  })
+}
+
+export const Assistant = () => {
+  const assistantId = env.OPENAI_ASSISTANT_ID
+  if (!assistantId) {
+    throw new Error("No assistant id found")
```

Comment:
we must not throw

Reply:
it's a misconfiguration if we ends up here. are you sure throw is not acceptable? I'm fine crashing the app if it's misconfigured / not supposed to work because we miss an env variable

Reply:
Throwing on illegal states at startup is good - don't want it to start running and silently fail when we could have been notified of the issue much earlier via CrashBackoffLoop.

Reply:
a point here is that it's not going to crash on startup as of now, but also crash when the Assistant() is been called. 

Alternatively, I can make the openai env var mandatory but this requires some synchronization in the rollout with the chart.

Reply:
we dont want to crash the entire pod when someone init a support request